pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaultevictor
+
+import (
+	// "context"
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+const (
+	PluginName            = "DefaultEvictor"
+	evictPodAnnotationKey = "descheduler.alpha.kubernetes.io/evict"
+)
+
+var _ framework.EvictorPlugin = &DefaultEvictor{}
+
+type constraint func(pod *v1.Pod) error
+
+// DefaultEvictor is the first EvictorPlugin, which defines the default extension points of the
+// pre-baked evictor that is shipped.
+// Even though we name this plugin DefaultEvictor, it does not actually evict anything,
+// This plugin is only meant to customize other actions (extension points) of the evictor,
+// like filtering, sorting, and other ones that might be relevant in the future
+type DefaultEvictor struct {
+	args        *DefaultEvictorArgs
+	constraints []constraint
+}
+
+// IsPodEvictableBasedOnPriority checks if the given pod is evictable based on priority resolved from pod Spec.
+func IsPodEvictableBasedOnPriority(pod *v1.Pod, priority int32) bool {
+	return pod.Spec.Priority == nil || *pod.Spec.Priority < priority
+}
+
+// HaveEvictAnnotation checks if the pod have evict annotation
+func HaveEvictAnnotation(pod *v1.Pod) bool {
+	_, found := pod.ObjectMeta.Annotations[evictPodAnnotationKey]
+	return found
+}
+
+type DefaultEvictorOptions struct {
+	Priority      *int32
+	NodeFit       bool
+	LabelSelector labels.Selector
+}
+
+// New builds plugin from its arguments while passing a handle
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	defaultEvictorArgs, ok := args.(*DefaultEvictorArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type defaultEvictorFilterArgs, got %T", args)
+	}
+	ev := &DefaultEvictor{
+		args: defaultEvictorArgs,
+	}
+
+	if ev.args.EvictFailedBarePods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
```

Comment:
`klog.V(1).InfoS("Warning: EvictFailedBarePods is set to True. This could cause eviction of pods without ownerReferences.")` log